wp_api/src/login.rs

```diff
@@ -66,6 +68,35 @@ pub struct WpApiDetails {
     pub authentication: WpApiDetailsAuthenticationMap,
     #[serde(default, deserialize_with = "deserialize_false_or_string")]
     pub site_icon_url: Option<String>,
+    pub routes: HashMap<String, WpRoute>,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, uniffi::Record)]
+pub struct WpRoute {
+    pub namespace: String,
+    pub methods: Vec<String>,
+    pub endpoints: Vec<WpEndpoint>,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, uniffi::Record)]
+pub struct WpEndpoint {
+    pub methods: Vec<String>,
+    #[serde(deserialize_with = "deserialize_empty_array_or_hashmap")]
+    pub args: HashMap<String, WpEndpointArg>,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, uniffi::Record)]
+pub struct WpEndpointArg {
+    pub required: bool,
+
+    pub default: Option<JsonValue>,
+    pub description: Option<String>,
+    #[serde(deserialize_with = "deserialize_string_vec_or_string_as_option")]
+    #[serde(default)]
+    pub r#type: Option<Vec<String>>,
+    pub r#enum: Option<Vec<String>>,
```

Comment:
What does the `r#` mean?

Reply:
`type` and `enum` are reserved works, so you need to do this to use them as field names